src/tests/multihost/ad/test_adparameters_ported.py

```diff
@@ -3535,3 +3536,136 @@ def test_0042_ad_parameters_nonroot_user_sssd(
+        ad_realm = multihost.ad[0].domainname.upper()
+
+        # Create AD user and group
+        (aduser, _) = create_aduser_group
+
+        # Configure sssd
+        client = sssdTools(multihost.client[0], multihost.ad[0])
+        client.backup_sssd_conf()
+
+        dom_section = f'domain/{client.get_domain_section_name()}'
+        sssd_params = {
+            'ad_domain': multihost.ad[0].domainname,
+            'debug_level': '0x4000',
+            'cache_credentials': 'True',
+            'ad_server': multihost.ad[0].hostname,
+            'krb5_store_password_if_offline': 'True',
+            'krb5_server': f'{multihost.ad[0].hostname}:6666',
+            'id_provider': 'ldap',
+            'auth_provider': 'krb5',
+            'access_provider': 'ad',
+            'krb5_realm': ad_realm,
+            'ldap_sasl_mech': 'GSSAPI',
+            'fallback_homedir': '/home/%u',
+        }
+        client.sssd_conf(dom_section, sssd_params)
+
+        # Forward ports on AD machine so 6666 works
+        multihost.ad[0].run_command(
+            "netsh interface portproxy add v4tov4 listenaddress=0.0.0.0"
+            " listenport=6666 connectaddress=127.0.0.1 connectport=88; "
+            " netsh interface portproxy show all",
+            raiseonerr=False
+        )
+
+        # Allow 6666 on firewall
+        fw_cmd = "powershell.exe -inputformat none -noprofile \"New-NetFire" \
+                 "wallRule -DisplayName 'alt-krb-Inbound' -Profile @(" \
+                 "'Domain', 'Private', 'Public') -Direction Inbound -Action " \
+                 "Allow -Protocol TCP -LocalPort 6666\""
+        multihost.ad[0].run_command(fw_cmd, raiseonerr=False)
+
+        # Workaround for DNS
+        multihost.client[0].run_command(
+            f'grep "{multihost.ad[0].hostname}" /etc/hosts || echo -n "'
+            f'\n{multihost.ad[0].ip} {multihost.ad[0].hostname}\n'
+            f'">> /etc/hosts',
+            raiseonerr=False
+        )
+        multihost.client[0].run_command(
+            'cat /etc/hosts',
+            raiseonerr=False
+        )
+
+        # Clear cache and restart SSSD
+        client.clear_sssd_cache()
+
+        # Debug
+        multihost.client[0].run_command(
+            'cat /etc/sssd/sssd.conf', raiseonerr=False)
+
+        # Run kinit for the user
+        kinit_cmd = multihost.client[0].run_command(
+            f'kinit {aduser}@{ad_realm}', stdin_text='Secret123',
+            raiseonerr=False)
+
+        # Run ssh
+        multihost.client[0].run_command(
+            f'ssh -o StrictHostKeychecking=no -o NumberOfPasswordPrompts=1 '
+            f'-o UserKnownHostsFile=/dev/null -l {aduser}@{ad_realm} '
+            f'localhost whoami',
+            stdin_text='Secret123',
+            raiseonerr=False)
+
+        multihost.client[0].run_command(
+            'cat  /var/lib/sss/pubconf/kdcinfo.*', raiseonerr=False)
+
+        # Give it some time sot the log can be written
+        time.sleep(30)
```

Comment:
Maybe we can remove the sleep or at least reduce it so that the test finishes faster.

Reply:
Done